flask_admin/model/widgets.py

```diff
@@ -61,3 +61,81 @@ def __call__(self, field, **kwargs):
         kwargs.setdefault('data-placeholder', placeholder)
 
         return HTMLString('<input %s>' % html_params(name=field.name, **kwargs))
+
+
+class XEditableWidget(object):
+    """
+        WTForms widget that provides in-line editing for the list view.
+        
+        Determines how to display the x-editable/ajax form based on the 
+        field inside of the FieldList (StringField, IntegerField, etc).
+    """
+    def __call__(self, field, **kwargs):
+        value = kwargs.pop("value", "")
+
+        kwargs.setdefault('data-role', 'x-editable')
+        kwargs.setdefault('data-url', './')
+
+        kwargs.setdefault('id', field.id)
+        kwargs.setdefault('name', field.name)
+        kwargs.setdefault('href', '#')
+
+        if not kwargs.get('pk'):
+            raise Exception('pk required')
+        kwargs['data-pk'] = str(kwargs.pop("pk"))
+
+        kwargs['data-csrf'] = kwargs.pop("csrf", "")
+
+        # get first entry from FieldList to determine field type
+        subfield = field.entries[0]
+
+        if subfield.type == 'StringField':
```

Comment:
What if someone is using a custom validator like "date must be greater than today"?

I left "field_args=self.form_args" out of scaffold_list_form: https://github.com/mrjoes/flask-admin/pull/756#discussion-diff-22463150R625

I figured it may cause some issues with how the field is displayed if the same form_args are applied, but I can test to see if this is the case.